src/approaches/pg3_approach.py

```diff
@@ -354,36 +371,44 @@ def _get_atom_plan_for_task(self, ldl: LiftedDecisionList,
     @functools.lru_cache(maxsize=None)
     def _get_demo_atom_plan_for_task(
```

Comment:
have you verified the caching works as intended?

Reply:
yeah it does